htdocs/core/modules/fichinter/mod_pacific.php

```diff
@@ -82,7 +83,7 @@ public function getExample()
 	 *  Checks if the numbers already in the database do not
 	 *  cause conflicts that would prevent this numbering working.
 	 *
-	 *  @param  Object		$object		Object we need next value for
+	 *  @param  Fichinter	$object		Object we need next value for
 	 *  @return boolean     			false if conflict, true if ok
 	 */
 	public function canBeActivated($object)
```

Comment:
@mdeweerd no phan warning here ? 🙄

Reply:
A reason could be that the file is in the baseline, but it is not.
Example:
https://github.com/frederic34/dolibarr/blob/40ff3ca17355720a8a8f8629248f39881ba40d18/dev/tools/phan/baseline.txt#L84

However, mod_pacific ultimately derives from CommonDocGenerator and canBeActivated is not defined in any parent class.

mod_asset_standard relies on CommonNumRefGenerator which defines canBeActivated with a typing already and therefore needs to be compatible.

Phan requires that the class type is a superclass, not a subclass which I find logical.

If a subclass is required than the first test the method would have to be (
```
if (!$object instanceof Asset) { return false; }
```
or 

```
if (!$object instanceof Asset) { return parent::canBeActivated($object); }
```

Reply:
_However, mod_pacific ultimately derives from CommonDocGenerator and canBeActivated is not defined in any parent class._
mod_pacific relies on CommonNumRefGenerator, not CommonDocGenerator

Reply:
class mod_pacific extends ModeleNumRefFicheinter
abstract class ModeleNumRefFicheinter extends CommonNumRefGenerator  // More than one class in same file (not expected)

And CommonNumRefGenerator has:
```php
    /**
     *  Checks if the numbers already in the database do not
     *  cause conflicts that would prevent this numbering working.
     *
     *  @param  CommonObject    $object Object we need next value for
     *  @return boolean                 false if conflict, true if ok
     */
    public function canBeActivated($object)
    {
        return true;
    }
```

Reply:
but there is no warning with mod_pacific, no warning for mod_holiday_madonna

I have warning for some case, mod_facture_terre

all object derives from CommonObject

Reply:
Ok, sorry.

I may try to look a bit deeper - it could be a phan bug.

I made a test case that does report the Issue, so it should be reported when the class specified in the overload method is more specific than the class in its parent method.
```
<?php

class CommonObject {
    /** @var bool */
    public $dummy = false;
}

class SpecificObject extends CommonObject {
}

class ParentA {
    /** @var bool */
    public $dummy = true;
    /**
     *  Check
     *
     *  @param  CommonObject    $object Object
     *  @return boolean                 false if conflict, true if ok
     */
    public function canBeActivated($object) {
        return $object->dummy;
    }
}

class ChildA extends ParentA {
    /**
     *  Check
     *
     *  @param  SpecificObject         $object Object
     *  @return boolean                 false if conflict, true if ok
     */
    public function canBeActivated($object) {
        return $object->dummy;
    }
}
```

Reply:
I generated the phan report locally, these case are in the report:

```
htdocs\core\modules\fichinter\mod_pacific.php:89 PhanParamSignatureMismatch Declaration of function canBeActivated(\Fichinter $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs\core\class\commonnumrefgenerator.class.php:138 (Expected \Fichinter $object to have the same type as \CommonObject $object or a supertype)
htdocs\core\modules\holiday\mod_holiday_madonna.php:83 PhanParamSignatureMismatch Declaration of function canBeActivated(\Holiday $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs\core\class\commonnumrefgenerator.class.php:138 (Expected \Holiday $object to have the same type as \CommonObject $object or a supertype)
```htdocs\core\modules\holiday\mod_holiday_madonna.php:83 PhanParamSignatureMismatch Declaration of function canBeActivated(\Holiday $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs\core\class\commonnumrefgenerator.class.php:138 (Expected \Holiday $object to have the same type as \CommonObject $object or a supertype)
```

Reply:
In fact, when we look at the raw log online, we see the same:

```
canBeActivated(\CommonObject $object) : bool defined in htdocs/core/class/commonnumrefgenerator.class.php:138 (Expected \Facture $object to have the same type as \CommonObject $object or a supertype)
2024-03-11T20:04:41.1111002Z ##[warning]mod_facture_terre.php: PhanParamSignatureMismatch: Declaration of function canBeActivated(\Facture $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs/core/class/commonnumrefgenerator.class.php:138 (Expected \Facture $object to have the same type as \CommonObject $object or a supertype)
2024-03-11T20:04:41.1116845Z ##[warning]mod_pacific.php: PhanParamSignatureMismatch: Declaration of function canBeActivated(\Fichinter $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs/core/class/commonnumrefgenerator.class.php:138 (Expected \Fichinter $object to have the same type as \CommonObject $object or a supertype)
2024-03-11T20:04:41.1122499Z ##[warning]mod_holiday_madonna.php: PhanParamSignatureMismatch: Declaration of function canBeActivated(\Holiday $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs/core/class/commonnumrefgenerator.class.php:138 (Expected \Holiday $object to have the same type as \CommonObject $object or a supertype)
2024-03-11T20:04:41.1128689Z ##[warning]mod_evaluation_standard.php: PhanParamSignatureMismatch: Declaration of function canBeActivated(\Evaluation $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs/core/class/commonnumrefgenerator.class.php:138 (Expected \Evaluation $object to have the same type as \CommonObject $object or a supertype)
2024-03-11T20:04:41.1134502Z ##[warning]mod_member_advanced.php: PhanParamSignatureMismatch: Declaration of function canBeActivated(\Adherent $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs/core/class/commonnumrefgenerator.class.php:138 (Expected \Adherent $object to have the same type as \CommonObject $object or a supertype)
2024-03-11T20:04:41.1140139Z ##[warning]mod_member_simple.php: PhanParamSignatureMismatch: Declaration of function canBeActivated(\Adherent $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs/core/class/commonnumrefgenerator.class.php:138 (Expected \Adherent $object to have the same type as \CommonObject $object or a supertype)
2024-03-11T20:04:41.1145669Z ##[warning]mod_mo_standard.php: PhanParamSignatureMismatch: Declaration of function canBeActivated(\MO $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs/core/class/commonnumrefgenerator.class.php:138 (Expected \MO $object to have the same type as \CommonObject $object or a supertype)
2024-03-11T20:04:41.1151243Z ##[warning]mod_lot_standard.php: PhanParamSignatureMismatch: Declaration of function canBeActivated(\ProductLot $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs/core/class/commonnumrefgenerator.class.php:138 (Expected \ProductLot $object to have the same type as \CommonObject $object or a supertype)
2024-03-11T20:04:41.1156903Z ##[warning]mod_sn_standard.php: PhanParamSignatureMismatch: Declaration of function canBeActivated(\ProductLot $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs/core/class/commonnumrefgenerator.class.php:138 (Expected \ProductLot $object to have the same type as \CommonObject $object or a supertype)
2024-03-11T20:04:41.1162571Z ##[warning]mod_project_simple.php: PhanParamSignatureMismatch: Declaration of function canBeActivated(\Project $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs/core/class/commonnumrefgenerator.class.php:138 (Expected \Project $object to have the same type as \CommonObject $object or a supertype)
2024-03-11T20:04:41.1168433Z ##[warning]mod_propale_marbre.php: PhanParamSignatureMismatch: Declaration of function canBeActivated(\Propal $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs/core/class/commonnumrefgenerator.class.php:138 (Expected \Propal $object to have the same type as \CommonObject $object or a supertype)
2024-03-11T20:04:41.1174184Z ##[warning]mod_reception_beryl.php: PhanParamSignatureMismatch: Declaration of function canBeActivated(\Reception $object) : bool should be compatible with function canBeActivated(\CommonObject $object) : bool defined in htdocs/core/class/commonnumrefgenerator.class.php:138 (Expected \Reception $object to have the same type as \CommonObject $object or a supertype)
2024-03-11T20:04:41.1180200Z ##[warning]mod_stocktransfer_standard.php: PhanParamSignatureMismatch: Declaration of function canBeActivated(\StockTransfer $object) : bool should be compatible with function
```

![image](https://github.com/Dolibarr/dolibarr/assets/1504752/45131caf-3c50-4f94-8943-8c832c58501c)